arrow/src/main/java/org/apache/calcite/adapter/arrow/ArrowTranslator.java

```diff
@@ -185,6 +187,34 @@ private String translateOp2(String op, String name, RexLiteral right) {
     return name + " " + op + " " + valueString + " " + valueType;
   }
 
+  /** Translates a call to a unary operator. */
+  private String translateUnary(String op, RexCall call) {
+    final RexNode opNode = call.operands.get(0);
+    @Nullable String expression = translateUnary2(op, opNode);
+
+    if (expression != null) {
+      return expression;
+    }
+
+    throw new UnsupportedOperationException("Unsupported unary operator " + call);
+  }
+
+  /** Translates a call to a unary operator. Returns null on failure. */
+  private @Nullable String translateUnary2(String op, RexNode opNode) {
+    if (opNode.getKind() == SqlKind.INPUT_REF) {
+      final RexInputRef inputRef = (RexInputRef) opNode;
+      final String name = fieldNames.get(inputRef.getIndex());
+      return translateUnaryOp(op, name);
+    }
+
+    return null;
+  }
+
+  /** Combines a field name and a unary operator to produce a predicate string. */
+  private String translateUnaryOp(String op, String name) {
```

Comment:
Nit: I just noticed this is used in a single place, I guess we can inline as it's a single line and the size would be almost the same.

Reply:
I think we'll need at least one more unary operator afterwards, which I'll also look into: [Support IS NOT NULL in Arrow adapter](https://issues.apache.org/jira/browse/CALCITE-6295?jql=project%20%3D%20CALCITE%20AND%20component%20%3D%20arrow-adapter), so I guess we can keep it? WDYT?